source/core/constraint/constraint.d

```diff
@@ -13,22 +11,61 @@ public enum ConstraintType : string
 
 public abstract class Constraint : IConstraint
 {
+
 	protected this(ConstraintType constraintType)
 	{
 		_constraintType = constraintType;
 	}
 
-	public abstract void connect(Cell cell);
+
+	///
 	public abstract bool isValid(in int digit);
 
+
+	public static void createSingle(ConstraintType type, Cell cell1, Cell cell2)
+	{
+		final switch(type)
+		{
+			case ConstraintType.Unique:
+				import core.constraint.unique : UniqueConstraint;
```

Comment:
This import is repeated three times; isn't it better to make it global?

Reply:
I changed this to follow the D style: https://dlang.org/dstyle.html#phobos_imports

Reply:
However if a function module is used, say in 80% or greater of the functions in a file, well, yes I might import it globaly. In this case it is only used in those three static functions for a very specific reason, call the correspondent static function of that module's class. In the future this will be changed to a mixin.